output/print_methods.go

```diff
@@ -2,59 +2,74 @@ package output
 
 import (
 	"fmt"
+	"net/http"
 	"strings"
 
 	"github.com/fatih/color"
 	"github.com/visola/go-http-cli/request"
 )
 
-type bodyBuffer struct {
-	*strings.Reader
-}
-
-func (bb *bodyBuffer) Close() error {
-	return nil
-}
-
 // PrintRequest outputs the http.Request
-func PrintRequest(request request.Request) error {
+func PrintRequest(request request.Request) {
 	boldGreen := color.New(color.Bold, color.FgGreen).PrintfFunc()
 	boldGreen("\n%s %s\n", request.Method, request.URL)
 
-	sentHeaderKeyColor := color.New(color.Bold, color.FgBlack).PrintfFunc()
-	sentHeaderValueColor := color.New(color.FgBlack).PrintfFunc()
+	printHeaders(request.Headers)
+	printCookies(request.Cookies)
+	printBody(request.Body, ">>")
+}
 
-	for headerName, values := range request.Headers {
-		sentHeaderKeyColor("%s:", headerName)
-		if len(values) > 1 {
-			for _, val := range values {
-				sentHeaderValueColor("\n  %s", val)
-			}
-			fmt.Println("")
-		} else {
-			sentHeaderValueColor(" %s\n", values[0])
+// PrintResponse outputs a http.Response
+func PrintResponse(response request.Response) {
+	if response.StatusCode < 300 {
+		color.Green("%s %s\n", response.Status, response.Protocol)
+	} else if response.StatusCode < 400 {
+		color.Yellow("%s %s\n", response.Status, response.Protocol)
+	} else {
+		color.Red("%s %s\n", response.Status, response.Protocol)
+	}
+
+	printHeaders(response.Headers)
+	printBody(response.Body, "<<")
+}
+
+func printBody(body string, linePrefix string) {
+	if body != "" {
```

Comment:
Does this check matter?

If `body` is `""` then `split` is `[]` and nothing will get printed, right?

If you did want to keep the check, should it live in `printBody` or should it live in `PrintRequest`? It feels a little funky to have the entire function inside a logical branch, but I don't have a strong argument aside from "my feelings".

Reply:
Well, that check avoids the tentative text processing and the creation of a new function to print the body in color.
And as you said, I see nothing wrong with a whole method inside a logical branch. I think it is the responsibility of the function to decide what to do if there's not body.

Reply:
hmmm I see your point - I guess my point was that the act of printing the body is different than the logic that determines *when to print the body*.

Imagine instead of just the one a `body != ""` case you had multiple other cases
```golang
func printBody(body string) {
  if body != "" {
  // don't print
  } else if body == "jae" {
  // print with some awesome format
  } else if body == "jaebradley" {
  // print some even more awesome format
 }
}
```
I would argue that the printing / formatting logic is separate from the condition of when to print.

```golang
func printRequestDetails(r Request) {
  if r.body != "" {
    // don't print
  } else if r.body == "jae" {
	printBody("something awesome")
  } else if body == "jaebradley" {
    printBody("something even more awesome")
  }
}
```

Reply:
> I guess my point was that the act of printing the body is different than the logic that determines when to print the body.

I think you're right. They are two different things and two different responsibilities. But for now, it's pretty contained and if I take it out of this method, it would be duplicated somewhere else. So I prefer to leave this way until I need another method: `shouldPrintBody` or something like that.